File: DocTok/ClientApp/src/services/DocumentService.ts
import customAxios from "../customAxios";
import IDocumentService from './interfaces/IDocumentService';
import DocumentPage from '../models/responseModels/DocumentPage';
import {DocumentPageRequestModel} from '../models/requestModels/Documents/DocumentPageRequestModel';
import { DocumentHierarchyModel } from "../models/responseModels/DocumentHierarchyModel";
import { DocumentPageDetailResponseModel } from "../models/responseModels/Detail/DocumentDetailResponseModel";

export default class DocumentService implements IDocumentService{

    public async getHierarchyByProjectId(id: number): Promise<DocumentHierarchyModel[]> {
        const response = await customAxios.get(`/documents/hierarchy/project/${id}`)
        return response.data
    }

    public async get(): Promise<DocumentPage[]> {
        const response = await customAxios.get('/documents')
        return response.data
    }

    public async getById(id: number): Promise<DocumentPageDetailResponseModel> {
        const response = await customAxios.get(`/documents/${id}`)
        return response.data
    }
    
    public async getByProjectId(id: number): Promise<DocumentHierarchyModel[]> {
        const response = await customAxios.get(`/documents/project/${id}`)
        return response.data
    }

    public async createDocument(documentRequestModel : DocumentPageRequestModel): Promise<DocumentPage>{
        const response = await customAxios.post(`/documents`,documentRequestModel)
        return response.data
    }

    public async updateDocument(documentRequestModel : DocumentPageRequestModel): Promise<DocumentPage>{
        const response = await customAxios.put(`/documents`,documentRequestModel)
        return response.data
    }

    public async deleteDocument(id: number) : Promise<void> {
        await customAxios.delete(`/documents/${id}`)
    }
}

File: DocTok.BusinessLayer/Services/DocumentService.cs
using DocTok.BusinessLayer.Interfaces;
using DocTok.DataAccess.Interfaces;
using DocTok.Shared.Entities;
using DocTok.Shared.RequestModels.Document;
using DocTok.Shared.ResponseModels.Document;
using DocTok.Shared.ResponseModels.Document.Detail;

namespace DocTok.BusinessLayer.Services
{
    public class DocumentService : IDocumentService
    {
        private readonly IDocumentRepository documentRepository;

        public DocumentService(IDocumentRepository documentRepository)
        {
            this.documentRepository = documentRepository;
        }

        public async Task<IEnumerable<Document>> Get()
        {
            return await documentRepository.Get();
        }

        public async Task<DocumentDetailResponseModel> GetById(int id)
        {
            return await documentRepository.GetById(id);
        }

        public async Task<IEnumerable<DocumentHierarсhyModel>> GetHierarchyByProjectId(int id)
        {
            return await documentRepository.GetHierarchyByProjectId(id);
        }

        public async Task<Document> Create(DocumentRequestModel document)
        {
            return await documentRepository.Create(document);
        }

        public async Task<Document> Update(DocumentRequestModel document)
        {
            return await documentRepository.Update(document);
        }

        public async Task Delete(int id)
        {
            await documentRepository.Delete(id);
        }

        public async Task<IEnumerable<DocumentHierarсhyModel>> GetByProjectId(int id)
        {
            return await documentRepository.GetByProjectId(id);
        }
    }
}
